config/v1/tests/apiservers.config.openshift.io/KMSEncryptionProvider.yaml

```diff
@@ -0,0 +1,31 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "APIServer"
+crdName: apiservers.config.openshift.io
+featureGate: KMSEncryptionProvider
+tests:
+  onCreate:
```

Comment:
Can you please add some cases that show the various validations returning errors so that we know that they are working please

Reply:
Added more tests to check all the validations.